webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TestRestFromFileDef.scala

```diff
@@ -82,6 +90,30 @@ class TestRestFromFileDef extends TraitTestApiFromYamlFiles with AfterAll {
     ("api_revisions.yml" -> copyTransformApiRevision _)
   )
 
+  // there is some tests that need change in data model, that we don't want to have in all plugins.
+  // For ex, the tests for inherited values
+  val badOverrideType: GlobalParameter =
+    GlobalParameter("badOverrideType", GitVersion.DEFAULT_REV, "a string".toConfigValue, None, "a string at first", None)
+  restTestSetUp.mockParameters.paramsRepo.paramsMap.update(m => m + (badOverrideType.name -> badOverrideType)).runNow
+
+  val gProp: GroupProperty = GroupProperty
+    .parse(
+      "badOverrideType",
+      GitVersion.DEFAULT_REV,
+      """{ "now":"a json" }""",
+      None,
+      None
+    )
+    .getOrElse(null) // for test
+  import com.softwaremill.quicklens.*
+
+  val g0Id: NodeGroupId = restTestSetUp.mockNodeGroups.g0.id
+  (for {
+    g <- restTestSetUp.mockNodeGroups.groupsRepo.getNodeGroupOpt(g0Id)(QueryContext.testQC).notOptional("test")
+    up = g._1.modify(_.properties).using(_.appended(gProp))
+    _ <- restTestSetUp.mockNodeGroups.groupsRepo.update(up, ModificationId("test"), eventlog.RudderEventActor, None)
+  } yield ()).runNow
```

Comment:
The by feature directory solution seems also a simpler solution, from our usage is seems more convenient than having to write imperative fixture setup code...
Also a more elaborate solution could be to parameterize all the mocks with the test data/fixtures. I am thinking about a structure which would make use of overrides like this :
```scala
class MockNode(nodeFixtures: NodeFixtures) {
  object nodeFactRepo {...}
}

// In Rudder
trait NodeFixtures {
  val nodeFacts: Map[NodeId, NodeFact] = ...
}
trait RestNodeFixtures extends RestNodeFixtures {
  // specific to rudder-rest
}

// In plugins
trait SystemUpdatesNodeFixtures extends NodeFixtures {
  // specific to rudder-rest
}
```
I think keeping a declarative approach for fixtures (with value overrides) would avoid the pain of understanding the imperative one

Reply:
it could/seems to be the correct long term solution (not sure why traits, thought, that looks like just plain old data at different level that just need to be given at class instanciation ; we could even just have them in the mock class object and use them as default value for the param. It's simple to change data in scala)